src/sentry/api/endpoints/group_details.py

```diff
@@ -334,9 +339,29 @@ def delete(self, request, group):
         ).update(status=GroupStatus.PENDING_DELETION)
         if updated:
             GroupHash.objects.filter(group=group).delete()
+
+            transaction_id = uuid4().hex
+            project = group.project
+
             delete_group.apply_async(
-                kwargs={'object_id': group.id},
+                kwargs={
+                    'object_id': group.id,
+                    'transaction_id': transaction_id,
+                },
                 countdown=3600,
             )
 
+            self.create_audit_entry(
+                request=request,
+                organization_id=project.organization_id if project else None,
```

Comment:
It's awkwardly technically possible that `group.project` may be `NULL`, so this just makes sure we don't blow up if that case were to happen.